source/common/network/utility.cc

```diff
@@ -619,16 +612,27 @@ Api::IoCallUint64Result Utility::readFromSocket(IoHandle& handle,
   }
 
   if (handle.supportsMmsg()) {
-    const uint32_t num_packets_per_mmsg_call = 16u;
-    const uint32_t num_slices_per_packet = 1u;
-    absl::FixedArray<Buffer::InstancePtr> buffers(num_packets_per_mmsg_call);
+    const auto max_packet_size = udp_packet_processor.maxPacketSize();
+
+    // Buffer::ReservationSingleSlice is always passed by value, and can only be constructed
+    // by Buffer::Instance::reserve(), so this is needed to keep a fixed array
+    // in which all elements are legally constructed.
+    struct BufferAndReservation {
+      BufferAndReservation(uint64_t max_packet_size)
+          : buffer_(std::make_unique<Buffer::OwnedImpl>()),
+            reservation_(buffer_->reserveSingleSlice(max_packet_size, true)) {}
+
+      Buffer::InstancePtr buffer_;
+      Buffer::ReservationSingleSlice reservation_;
+    };
+    constexpr uint32_t num_packets_per_mmsg_call = 16u;
```

Comment:
We're reserving 16 slices in a loop here.  Increasing the freelist size to 16 would be helpful.

Reply:
The freelist only allows default_slice_size_ slices into it. These are probably not that size; most likely they're 4k slices.

It may make sense to also have a freelist for udp packets, but in theory, there could be multiple udp packet sizes on different listeners. Possibly the freelist could be on the `UdpPacketProcessor` itself.